app/src/main/java/com/github/se/project/model/lesson/LessonViewModel.kt

```diff
@@ -14,15 +14,20 @@ import kotlinx.coroutines.flow.asStateFlow
  * ViewModel for managing lessons and interacting with the LessonRepository. Handles the retrieval,
  * addition, and deletion of lessons.
  */
-class LessonsViewModel(private val repository: LessonRepository) : ViewModel() {
-  private val lessons_ = MutableStateFlow<List<Lesson>>(emptyList())
-  val lessons: StateFlow<List<Lesson>> = lessons_.asStateFlow()
+class LessonViewModel(private val repository: LessonRepository) : ViewModel() {
```

Comment:
In the LessonRepository class, there is some function that updates a Lesson. This updateLesson function will also need to be implemented here so that we can use it in our program. 
Maybe it was intended that it is not yet implemented because for now we have no screen to update the lesson but it should me made at some point in the project I think.